google-api-client/src/test/java/com/google/api/client/googleapis/MtlsTransportBaseTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.googleapis;
+
+import com.google.api.client.googleapis.util.MtlsUtils;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.util.SecurityUtils;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public abstract class MtlsTransportBaseTest  {
```

Comment:
We'll want to add more tests for the failure cases (when loadDefaultKeyStore() can fail) and how the `newTrustedTransport()` behaves in those cases.

Reply:
We can probably also add a test for the parameterless `newTrustedTransport` and use [`Assume.assumeTrue`](https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html#assumeTrue(boolean)) to protected the tests (skip if the test is not run under certain conditions) and we can possibly set up additional CI runs that configure mTLS on that test runner.

Reply:
added test for `newTrustedTransport` where `loadKeyStore` fails.

The parameterless `newTrustedTransport` tests with CI configuration I think we can add it later.